File: src/main/java/com/graqr/threshr/model/redsky/product/ProductDescription.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record ProductDescription(
        @JsonProperty("bullet_descriptions") List<String> bulletDescriptions,
        @JsonProperty("downstream_description") String downstreamDescription,
        @JsonProperty("soft_bullet_description") String softBulletDescription,
        @JsonProperty("soft_bullets") SoftBullets softBullets, String title) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/ProductDescription.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class ProductDescription(
    @param:JsonProperty("bullet_descriptions") val bulletDescriptions: List<String>,
    @param:JsonProperty("downstream_description") val downstreamDescription: String,
    @param:JsonProperty("soft_bullet_description") val softBulletDescription: String,
    @param:JsonProperty("soft_bullets") val softBullets: SoftBullets,
    val title: String
) {}
